projects/ngneat/hotkeys/src/lib/hotkeys.service.ts

```diff
@@ -0,0 +1,73 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+import { EventManager } from '@angular/platform-browser';
+import { Observable } from 'rxjs';
+
+interface Options {
+  element: HTMLElement;
+  trigger: 'keydown' | 'keyup';
+  description: string;
+  showInHelp: boolean;
+  group: string;
+}
+
+export type Hotkey = Partial<Options> & { keys: string };
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HotkeysService {
+  private readonly hotkeys = new Map();
+  private readonly defaults: Options = {
+    trigger: 'keydown',
+    element: this.document.documentElement,
+    description: '',
+    group: '',
+    showInHelp: true
+  };
+
+  constructor(private eventManager: EventManager, @Inject(DOCUMENT) private document: Document) {}
+
+  getShortcutsGroups(): string[] {
+    // map to get group name and remove duplicates
+    return Array.from(this.hotkeys)
+      .map(m => m[1])
+      .map(h => h.group)
+      .filter((el, idx, array) => array.indexOf(el) === idx);
+  }
+
+  getShortcutsByGroup(group: string): Hotkey[] {
+    // filter by group
+    return Array.from(this.hotkeys)
+      .map(m => m[1])
+      .filter(h => h.group === group)
+      .filter((el, idx, array) => array.indexOf(el) === idx);
+  }
+
+  getShortcuts(): Hotkey[] {
+    return Array.from(this.hotkeys).map(m => m[1]);
+  }
+
+  addShortcut(options: Hotkey): Observable<KeyboardEvent> {
+    const merged = { ...this.defaults, ...options };
+
+    if (this.hotkeys.has(merged.keys)) {
```

Comment:
I'm not sure we need this protection. If you think so, it doesn't make sense to return `null` when it expects to an observable.